docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -96,63 +116,42 @@ Example
     covariates = pd.read_csv(covariates_csv, index_col='sample_id')
     covariates = (covariates - covariates.mean())/covariates.std(ddof=0)
 
-------------------------
-Genotype matrix blocking
-------------------------
-
-``glow.wgr.functions.block_variants_and_samples`` creates two objects: a block genotype matrix and a sample block
-mapping.
+---------------------------------
+Stage 1. Genotype matrix blocking
+---------------------------------
+The first stage of GloWGR is to generate the block genotype matrix. The ``glow.wgr.functions.block_variants_and_samples`` function is used for this purpose and creates two objects: a block genotype matrix and a sample block mapping.
 
 Parameters
 ==========
 
-- ``genotypes``: Genotype DataFrame created by reading from any variant datasource supported by Glow, such as VCF. Must
-  also include a column ``values`` containing a numeric representation of each genotype.
-- ``sample_ids``: List of sample IDs. Can be created by applying ``glow.wgr.functions.get_sample_ids`` to a genotype
-  DataFrame.
-- ``variants_per_block``: Number of variants to include per block. We recommend 1000.
+- ``genotypes``: Genotype DataFrame including the ``values`` column generated as explained :ref:`above <genotype-data>`
+- ``sample_ids``: A python List of sample IDs. Can be created by applying ``glow.wgr.functions.get_sample_ids`` to a genotype
+  DataFrame
+- ``variants_per_block``: Number of variants to include in each block. We recommend 1000.
 - ``sample_block_count``: Number of sample blocks to create. We recommend 10.
 
 Return
 ======
 
 The function returns a block genotype matrix and a sample block mapping.
 
-.. warning::
+    - **Block genotype matrix**: The block genotype matrix can be conceptually imagined as an :math:`N \times M` matrix :math:`X` where each row represents an individual sample, and each column represents a variant, and each cell :math:`(i, j)` contains the genotype value for sample :math:`i` at variant :math:`j`.  Then imagine a coarse grid is laid on top of matrix :math:`X` such that matrix cells within the same coarse grid cell are all assigned to the same block. Each block :math:`x` is indexed by a sample block ID (corresponding to a list of rows belonging to the block) and a header block ID (corresponding to a list of columns belonging to the block). The sample block IDs are generally just integers 0 through the number of sample blocks. The header block IDs are strings of the form 'chr_C_block_B', which refers to the Bth block on chromosome C. The Spark DataFrame representing this block matrix can be thought of as the transpose of each block, i.e., :math:`x^T`, all stacked one atop another. Each row in the DataFrame represents the values from a particular column of :math:`X` for the samples corresponding to a particular sample block. The fields in the DataFrame and their content for a given row are as follows:
```

Comment:
This is a good description, but it would be much easier to understand in a figure. Maybe there was something in @LelandBarnard 's deck that we could adapt?

Reply:
Made and added a figure.